server.js

```diff
@@ -0,0 +1,125 @@
+// Requirements
+const knex = require('knex');
+const express = require('express');
+const bcrypt = require('bcryptjs');
+const knexConfig = require('./knexfile');
+const cors = require('cors');
+
+//change to token
+const jwt = require('jsonwebtoken');
+
+// Instantiations
+const server = express();
+const db = knex(knexConfig.development);
+
+// Middleware
+
+//change to token
+const secret = "lovey lovebirds"
+   function generateToken(user) {
+    const payload = {
+        username: user.username
+    };
+    const options = {
+        expiresIn: '1h',
+        jwtid: '12345',
+    }
+    return jwt.sign(payload, secret, options)
+}
+
+
+
+server.use(express.json());
+server.use(cors());
+
+function restricted(req, res, next) {
+
+    const token = req.headers.authorization;
+
+    if (token) {
+        jwt.verify(token, secret, (err, decodedToken) => {
+            if(err) {
+                res.status(401).json({message: "token invalid"})
+            } else {
+                req.user = {username: decodedToken.username};
+                next();
+        }
+    })
+    } else {
+    res.status(401).json({message: "no token"})
+    }
+}
+
+// Endpoints
+// server.get('/', (req, res) => {
+//     res.status(200).send('Server is running!');
+// });
+
+server.get('/', (req, res) => {
+    req.token.name = 'auth ii project';
```

Comment:
Ooo, what's this?

Reply:
mm, I thought I could name it anything? 😅